lib/resources/aws/aws_config_recorder.rb

```diff
@@ -50,35 +50,34 @@ def validate_params(raw_params)
       allowed_scalar_type: String,
     )
 
-    # Must give it a recorder_name
-    if validated_params[:recorder_name].nil?
-      raise ArgumentError, 'You must provide recorder_name to aws_config_recorder'
-    end
-
     validated_params
   end
 
   def fetch_from_api
     backend = BackendFactory.create(inspec_runner)
-    @query = { configuration_recorder_names: [@recorder_name] }
+    query = @recorder_name ? { configuration_recorder_names: [@recorder_name] } : {}
+
+    response = nil
+    begin
+      response = backend.describe_configuration_recorders(query)
+    rescue Aws::ConfigService::Errors::NoSuchConfigurationRecorderException
+      @exists = false
+      return
+    end
```

Comment:
Same as the other `fetch_from_api`, re: `rescue` at method level instead of introducing a `begin` block.